packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -151,76 +152,103 @@ export class BlockSwitcher extends Component {
 						{ ( hasBlockStyles ||
 							possibleBlockTransformations.length !== 0 ) && (
 							<div className="block-editor-block-switcher__container">
-								{ hasBlockStyles && (
-									<MenuGroup>
-										<div className="block-editor-block-switcher__label">
-											{ __( 'Styles' ) }
-										</div>
-										<BlockStyles
-											clientId={ blocks[ 0 ].clientId }
-											onSwitch={ onClose }
-											onHoverClassName={
-												this.onHoverClassName
-											}
-										/>
-									</MenuGroup>
-								) }
 								{ possibleBlockTransformations.length !== 0 && (
-									<MenuGroup>
-										<div className="block-editor-block-switcher__label">
-											{ __( 'Transform to' ) }
-										</div>
-										<BlockTypesList
-											items={ possibleBlockTransformations.map(
-												( destinationBlockType ) => ( {
-													id:
-														destinationBlockType.name,
-													icon:
-														destinationBlockType.icon,
-													title:
-														destinationBlockType.title,
-												} )
+									<NavigableToolbar
+										orientation="vertical"
+										className="block-editor-block-switcher-toolbar"
+										role="menu"
+										/* translators: accessibility text for the block toolbar */
+										aria-label={ __( 'Block switcher' ) }
+									>
+										<MenuGroup>
+											<div className="block-editor-block-switcher__label">
+												{ __( 'Transform to' ) }
+											</div>
+											<BlockTypesList
```

Comment:
Now that this list is different than the list of items displayed in the inserter through `BlockTypesList`, it might be more convenient to avoid all the logic to style it to fit this space by iterating over the list of transforms and using `MenuItem` component for each of them.

Reply:
I believe this could be better suited in a new PR with the combining of the two lists ( Blocks and Styles ), as there will be a lot of Component changing/moving around there. But others can provide their input as well..